crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1195,6 +1195,43 @@ def restart_remote_resources(controller, node)
         )
       end
 
+      # Move the upgraded node to ready state and wait until nova-compute resource
+      # is reported as running.
+      def start_remote_resources(controller, hostname)
+        save_node_action("starting services at remote node #{hostname}")
+        out = controller.run_ssh_cmd(
+          "crm --wait node ready remote-#{hostname}",
+          "60s"
+        )
+        unless out[:exit_code].zero?
+          raise_node_upgrade_error(
+            "Starting remote services at '#{hostname}' node has failed. " \
+            "Check /var/log/pacemaker.log at '#{controller.name}' and " \
+            "'#{hostname}' nodes for possible causes."
+          )
+        end
+        seconds = timeouts[:wait_until_compute_started]
+        begin
+          Timeout.timeout(seconds) do
+            loop do
+              out = controller.run_ssh_cmd(
+                "source /root/.openrc; " \
+                "openstack compute service list --insecure " \
+                "--host #{hostname} --service nova-compute -f value -c State"
+              )
+              break if !out[:stdout].nil? && out[:stdout].chomp == "up"
+            end
```

Comment:
Is it worth putting at least a small sleep in here (1 or 2 seconds), so that we don't totally hammer the controller?  Maybe not, just an idea.